fml/src/main/java/net/minecraftforge/fml/common/discovery/JarDiscoverer.java

```diff
@@ -59,6 +63,22 @@
                 FMLLog.fine("The mod container %s appears to be missing an mcmod.info file", candidate.getModContainer().getName());
                 mc = MetadataCollection.from(null, "");
             }
+            
+            if (atr!=null && atr.get("TweakClass") != null && atr.get("TweakName") != null)
+            {
+                // Tweak Mod with TweakName = ModContainer
+                String name = atr.get("TweakName");
```

Comment:
You're setting this to a String, so you should use `getValue` instead. `get` returns an object.